unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/HighlightsSubSection/HighlightsSubSectionMenu/HighlightsSubSectionComponentController.cs

```diff
@@ -130,63 +134,66 @@ internal void RequestAllPlacesAndEvents()
+        (IReadOnlyList<PlaceInfo> pageTwo, _ ) = await placesAPIService.GetMostActivePlaces(1, PAGE_SIZE, ct);
+        placesFromAPI.Clear();
+        placesFromAPI.AddRange(pageOne);
+        placesFromAPI.AddRange(pageTwo);
+
+        eventsAPIApiController.GetAllEvents(
+        OnSuccess: eventList =>
+        {
+            eventsFromAPI = eventList;
+            OnRequestedPlacesAndEventsUpdated();
+        },
+        OnFail: error =>
+        {
+            OnRequestedPlacesAndEventsUpdated();
+            Debug.LogError($"Error receiving events from the API: {error}");
+        });
     }
 
     internal void OnRequestedPlacesAndEventsUpdated()
     {
         friendsTrackerController.RemoveAllHandlers();
 
-        List<PlaceCardComponentModel> trendingPlaces = PlacesAndEventsCardsFactory.ConvertPlaceResponseToModel(FilterTrendingPlaces());
+        List<PlaceCardComponentModel> trendingPlaces = PlacesAndEventsCardsFactory.ConvertPlaceResponseToModel(placesFromAPI, DEFAULT_NUMBER_OF_TRENDING_PLACES);
         List<EventCardComponentModel> trendingEvents = PlacesAndEventsCardsFactory.CreateEventsCards(FilterTrendingEvents(trendingPlaces.Count));
         view.SetTrendingPlacesAndEvents(trendingPlaces, trendingEvents);
 
         view.SetFeaturedPlaces(PlacesAndEventsCardsFactory.ConvertPlaceResponseToModel(FilterFeaturedPlaces()));
         view.SetLiveEvents(PlacesAndEventsCardsFactory.CreateEventsCards(FilterLiveEvents()));
     }
 
-    internal List<PlaceInfo> FilterTrendingPlaces() => placesFromAPI.Take(DEFAULT_NUMBER_OF_TRENDING_PLACES).ToList();
     internal List<EventFromAPIModel> FilterLiveEvents() => eventsFromAPI.Where(x => x.live).Take(DEFAULT_NUMBER_OF_LIVE_EVENTS).ToList();
     internal List<EventFromAPIModel> FilterTrendingEvents(int amount) => eventsFromAPI.Where(e => e.highlighted).Take(amount).ToList();
     internal List<PlaceInfo> FilterFeaturedPlaces()
     {
-        List<PlaceInfo> featuredPlaces;
-
-        if (placesFromAPI.Count >= DEFAULT_NUMBER_OF_TRENDING_PLACES)
+        if (placesFromAPI.Count > DEFAULT_NUMBER_OF_TRENDING_PLACES)
         {
             int numberOfPlaces = placesFromAPI.Count >= (DEFAULT_NUMBER_OF_TRENDING_PLACES + DEFAULT_NUMBER_OF_FEATURED_PLACES)
                 ? DEFAULT_NUMBER_OF_FEATURED_PLACES
                 : placesFromAPI.Count - DEFAULT_NUMBER_OF_TRENDING_PLACES;
 
-            featuredPlaces = placesFromAPI
+            return placesFromAPI
                             .GetRange(DEFAULT_NUMBER_OF_TRENDING_PLACES, numberOfPlaces)
                             .ToList();
         }
-        else if (placesFromAPI.Count > 0)
-            featuredPlaces = placesFromAPI.Take(DEFAULT_NUMBER_OF_FEATURED_PLACES).ToList();
-        else
-            featuredPlaces = new List<PlaceInfo>();
 
-        return featuredPlaces;
+        if (placesFromAPI.Count > 0)
+            return placesFromAPI.Take(DEFAULT_NUMBER_OF_FEATURED_PLACES).ToList();
+
+        return new List<PlaceInfo>();
```

Comment:
I highly recommend to return `IReadOnlyList` and use `Array.Empty` instead of creating a new instance every time